Cesium3DTiles/src/Tileset.cpp

```diff
@@ -1201,12 +1197,28 @@ static bool isVisibleFromCamera(
  * @param fogDensity The fog density
  * @return Whether the tile is visible in the fog
  */
-static bool isVisibleInFog(double distance, double fogDensity) {
-  if (fogDensity <= 0.0) {
-    return true;
+static bool isVisibleInFog(
+    const std::vector<ViewState>& frustums,
+    const std::vector<double>& distances,
+    const std::vector<FogDensityAtHeight>& fogDensityTable) {
+
+  for (size_t i = 0; i < frustums.size() && i < distances.size(); ++i) {
+    const ViewState& frustum = frustums[i];
+    const double& distance = distances[i];
+
+    double fogDensity = computeFogDensity(fogDensityTable, frustum);
```

Comment:
this may re-compute fog density for every tile even though fog-density only change between frustums